substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/graal/meta/SubstrateBasicLoweringProvider.java

```diff
@@ -133,11 +135,19 @@ protected ObjectLayout getObjectLayout() {
     public void lower(Node n, LoweringTool tool) {
         if (n instanceof AssertionNode) {
             lowerAssertionNode((AssertionNode) n);
+        } else if (n instanceof ArrayIndexOfDispatchNode) {
```

Comment:
Could be moved to DefaultJavaLoweringProvider.java and have the foreign call retrieved like AMD64LoweringProviderMixin.java (and delete the mixin code)

Reply:
Done!